src/core/Resources.cpp

```diff
@@ -224,30 +224,31 @@ Resources::Resources()
 {
     const QString appDirPath = QCoreApplication::applicationDirPath();
 #if defined(Q_OS_UNIX) && !(defined(Q_OS_MACOS) && defined(WITH_APP_BUNDLE))
-    testResourceDir(KEEPASSX_DATA_DIR) || testResourceDir(QStringLiteral("%1/../%2").arg(appDirPath, KEEPASSX_DATA_DIR))
-        || testResourceDir(QStringLiteral("%1/%2").arg(KEEPASSX_PREFIX_DIR, KEEPASSX_DATA_DIR));
+    trySetResourceDir(KEEPASSX_DATA_DIR)
+        || trySetResourceDir(QStringLiteral("%1/../%2").arg(appDirPath, KEEPASSX_DATA_DIR))
+        || trySetResourceDir(QStringLiteral("%1/%2").arg(KEEPASSX_PREFIX_DIR, KEEPASSX_DATA_DIR));
 #elif defined(Q_OS_MACOS) && defined(WITH_APP_BUNDLE)
-    testResourceDir(appDirPath + QStringLiteral("/../Resources"));
+    trySetResourceDir(appDirPath + QStringLiteral("/../Resources"));
 #elif defined(Q_OS_WIN)
-    testResourceDir(appDirPath + QStringLiteral("/share"));
+    trySetResourceDir(appDirPath + QStringLiteral("/share"));
 #endif
 
     if (m_dataPath.isEmpty()) {
         // Last ditch check if we are running from inside the src or test build directory
-        testResourceDir(appDirPath + QStringLiteral("/../../share"))
-            || testResourceDir(appDirPath + QStringLiteral("/../share"))
-            || testResourceDir(appDirPath + QStringLiteral("/../../../share"));
+        trySetResourceDir(appDirPath + QStringLiteral("/../share"))
+            || trySetResourceDir(appDirPath + QStringLiteral("/../../share"));
```

Comment:
Could also be a format expression. Causes a bit more runtime overhead, but is a bit more explicit what the final string looks like.

Reply:
eh for this purpose its clear